pyo3-macros-backend/src/module.rs

```diff
@@ -261,9 +262,13 @@ fn module_initialization(options: PyModuleOptions, ident: &syn::Ident) -> TokenS
 
 /// Finds and takes care of the #[pyfn(...)] in `#[pymodule]`
 fn process_functions_in_module(options: &PyModuleOptions, func: &mut syn::ItemFn) -> Result<()> {
-    let mut stmts: Vec<syn::Stmt> = Vec::new();
     let krate = get_pyo3_crate(&options.krate);
 
+    let mut stmts: Vec<syn::Stmt> = vec![syn::parse_quote!(
+        #[allow(unknown_lints, unused_imports, redundant_imports)]
+        use #krate::{PyNativeType, types::PyModuleMethods};
```

Comment:
I'm hesitant to add this kind of `use` statement, which transparently injects things into the namespace of user code and so would seem to violate hygiene (see discussion in the follow-up PR here: https://github.com/PyO3/pyo3/pull/3899#discussion_r1503181871). An alternative would be to use fully-qualified references to trait methods in the macro-generated code. But as @Icxolu noted in that discussion, the patterns around these new `*Methods` traits are not very well established, so I could be persuaded :-)

Reply:
Yeah, after that discussion I tend to agree, that we should not inject these traits silently. If we still want to do it we should probably at least import them anonymously e.g. as `use "Trait" as _`. I would not consider it a blocker here, we can change in a followup if needed. Another thing would be if we should seal these traits, but that's another discussion entirely.

Reply:
Agreed. Maybe a good solution here it to write a private trait in `impl_/pymodule.rs` which has just the methods we need implemented for both `Bound<'_, PyModule>` and `&'py PyModule`, which we can then call directly without imports.

(This is actually probably very similar to what @LilyFoote had already done with the `#[doc(hidden)] fn wrap_pyfunction`, just located as a private trait inside the impl codebase.)